util/test/activate_chpl_test_venv.py

```diff
@@ -16,46 +16,56 @@ def error(message):
 def activate_venv():
     venv_dir = chpl_home_utils.get_chpl_test_venv()
     default_venv_dir = chpl_home_utils.get_chpl_venv()
 
     # user asserts that system already has the required dependencies installed:
     if venv_dir == 'none':
-        log('Skipping virtualenv activation because CHPL_TEST_VENV_DIR={0}. '
+        log('Skipping venv activation because CHPL_TEST_VENV_DIR={0}. '
             'test-venv requirements must be available.'.format(venv_dir))
 
     else:
         # using custom venv, does not check that our test requirements are met
         if venv_dir != default_venv_dir:
-            log('Using custom virtualenv because CHPL_TEST_VENV_DIR={0}. '
+            log('Using custom venv because CHPL_TEST_VENV_DIR={0}. '
                 'test-venv requirements must be available'.format(venv_dir))
 
         # check Chapel test-venv for successful installation sentinel
         else:
             sentinel_file = os.path.join(venv_dir, 'chpl-test-reqs')
             if not os.path.isfile(sentinel_file):
-                error('Chapel test virtualenv not available, run a top-level '
+                error('Chapel test venv not available, run a top-level '
                       '`{0} test-venv`'.format(chpl_make.get()))
 
-        activation_file = os.path.join(venv_dir, 'bin', 'activate_this.py')
-        if not os.path.isfile(activation_file):
-            error('Activation file {0} is missing'.format(activation_file))
+        # "activate" the venv
+        # this is based upon activate_this.py from virtualenv.
+        old_os_path = os.environ.get('PATH', '')
 
-        old_path = os.environ.get('PATH', '')
-
-        # actually activate
-        with open(activation_file) as f:
-            code = compile(f.read(), activation_file, 'exec')
-            exec(code, dict(__file__=activation_file))
-
-        # put PATH back so that the system python may be used.
+        # do not change PATH so that the system python may be used.
         # python scripts using the virtualenv dependencies should use
         #
         #   from test import activate_chpl_test_venv
         #
         # near to the 'import' of a virtual dependency.
-        os.environ['PATH'] = old_path
+
+        #os.environ['PATH'] = os.path.dirname(os.path.abspath(__file__)) + os.pathsep + old_os_path
+
+        base = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        site_packages = os.path.join(base, 'lib', 'python%s' % sys.version[:3], 'site-packages')
+
+        prev_sys_path = list(sys.path)
+        import site
+        site.addsitedir(site_packages)
+        sys.real_prefix = sys.prefix
+        sys.prefix = base
+        # Move the added items to the front of the path:
+        new_sys_path = []
+        for item in list(sys.path):
+            if item not in prev_sys_path:
+                new_sys_path.append(item)
+                sys.path.remove(item)
+        sys.path[:0] = new_sys_path
```

Comment:
All of this seems somewhat fragile to me. Can we open up an issue to investigate better alternatives? I'm wondering if we can create a wrapper that does `python3 -m venv start_test` or something. (No idea if that sort of syntax/style actually works, just wondering if nowadays we should do something different than activating once we've already started a script)